src/loadPage.ts

```diff
@@ -2,50 +2,53 @@ import path from 'path';
 import { existsSync } from 'fs';
 import { requireAsIfOnServer } from './server';
 import type { ExtendedOptions, PageFile } from './commonTypes';
+import { InternalError } from './_error/error';
 
 export function loadFile<FileType>({
   absolutePath,
+  nonIsolatedModules,
 }: {
   absolutePath: string;
+  nonIsolatedModules: string[];
 }): PageFile<FileType> {
   return {
     client: require(absolutePath),
-    server: requireAsIfOnServer<FileType>(absolutePath),
+    server: requireAsIfOnServer<FileType>({
+      path: absolutePath,
+      nonIsolatedModules,
+    }),
   };
 }
 
-export function loadPage<FileType>({
-  pagePath,
-  options,
-}: {
+type LoadPageOptions = {
   pagePath: string;
   options: ExtendedOptions;
-}): PageFile<FileType> {
-  const { pagesDirectory, pageExtensions } = options;
+};
+
+export function loadPage<FileType>({
+  pagePath,
+  options: { pageExtensions, pagesDirectory, nonIsolatedModules },
+}: LoadPageOptions): PageFile<FileType> {
   // @NOTE Here we have to remove pagePath's leading "/"
   const absolutePath = path.resolve(pagesDirectory, pagePath.substring(1));
 
   for (const pageExtension of pageExtensions) {
     const pathWithExtension = absolutePath + `.${pageExtension}`;
     if (existsSync(pathWithExtension)) {
-      return loadFile({ absolutePath: pathWithExtension });
+      return loadFile({ absolutePath: pathWithExtension, nonIsolatedModules });
     }
   }
 
-  throw new Error(
-    "[next-page-tester] Couldn't find required page file with matching extension"
+  throw new InternalError(
+    "Couldn't find required page file with matching extension"
   );
 }
 
-export function loadPageIfExists<FileType>({
-  pagePath,
-  options,
-}: {
-  pagePath: string;
-  options: ExtendedOptions;
-}): PageFile<FileType> | undefined {
+export function loadPageIfExists<FileType>(
+  options: LoadPageOptions
```

Comment:
I've stumbled twice upon this `options` types which sounds like our `options` object but it's not. On one side we avoid repetition, on the other types are less explicit. We'll see further on... :)